### ISSUE TITLE
Cannot use Redis on Azure with SSL

### ISSUE THREAD

**Issue by jmichelgarcia** [cloudfoundry/java-buildpack-auto-reconfiguration#66]
Hi everyone,

I have a Spring application deployed into Pivotal Cloud Foundry bound to a redis service. **What is the best way to force the usage of the SSL port** ? Azure disables non SSL access by default, check [here](https://docs.microsoft.com/en-us/azure/redis-cache/cache-configure#access-ports) but keeps sending a redisUrl with the non-ssl port.

How can I change the port after the auto-reconfiguration magic happens?

my VCAP is the following :

```
"azure-rediscache": [
   {
    "binding_name": null,
    "credentials": {
     "hostname": "xxx.redis.cache.windows.net",
     "name": "xxx",
     "port": 6379,
     "primaryKey": "xxxx",
     "redisUrl": "redis://xxx5e7:xxxDg%3D@xxxx.redis.cache.windows.net:6379",
     "secondaryKey": "xxxxx",
     "sslPort": 6380
    },
    "instance_name": "my-redis",
    "label": "azure-rediscache",
    "name": "my-redis",
    "plan": "basicc0",
    "provider": null,
    "syslog_drain_url": null,
    "tags": [
     "Azure",
     "Redis",
     "Cache"
 ...
```

**Comment by nebhale** [comment]
Java Buildpack Auto-reconfiguration builds on top of [Spring Cloud Connectors](https://cloud.spring.io/spring-cloud-connectors/) which makes the actual configuration decisions.  Auto-reconfiguration simply takes the defaults that SCC provides and replaces any existing types with them.  For this reason (among others) it is not recommended that you use Auto-reconfiguration in a production environment.

Instead, you should use Spring Cloud Connectors directly, [configuring the `ConnectionFactory`](http://cloud.spring.io/spring-cloud-connectors/spring-cloud-spring-service-connector.html#_redis) appropriately.  A quick look over the API though, indicates that you might not be able to override the port/secure-ness of the URI.  @scottfrederick would you agree with this assessment?

I think that the correct course of action though is for the Azure Service Broker to be updated so that it no longer suggests a URI that cannot be used.  You should [open an issue](https://github.com/Azure/meta-azure-service-broker) against that project for this change.

**Comment by scottfrederick** [comment]
An issue has been opened for this in the Azure Service Broker repo: https://github.com/Azure/meta-azure-service-broker/issues/166.

There is also a related issue in the Spring Cloud Connectors repo: https://github.com/spring-cloud/spring-cloud-connectors/issues/235.

**Comment by scottfrederick** [comment]
@nebhale This should be fixed with the latest versions of the service broker and Spring Cloud Connectors [2.0.3.RELEASE](https://github.com/spring-cloud/spring-cloud-connectors/releases/tag/v2.0.3.RELEASE).

**Comment by nebhale** [comment]
Resolved by 9745683

**Comment by jmichelgarcia** [comment]
Thank you !
